pkg/gotenberg/supervisor.go

```diff
@@ -212,18 +215,29 @@ func (s *processSupervisor) Run(ctx context.Context, logger *zap.Logger, task fu
 					}
 				}
 
+				err := s.runWithDeadline(ctx, task)
+
 				if s.maxReqLimit > 0 && s.reqCounter.Load() >= s.maxReqLimit {
-					s.logger.Debug("max request limit reached, restarting...")
-					err := s.runWithDeadline(ctx, func() error {
-						return s.restart()
-					})
-					if err != nil {
-						return fmt.Errorf("process restart before task: %w", err)
-					}
+					s.logger.Debug("max request limit reached, restarting eagerly...")
+
+					// mutexChan will be released manually (not as part of the defer) after the restart,
+					// because restart needs the lock but is happening in the background.
```

Comment:
It's actually confusing more than anything lol

Better to remove this comment altogether, as the code base makes sense without it anyway! 😉